File: KadaneAlgorithm/kadane.go
package main

import (
	"fmt"
)

func kadanesAlgorithm(nums []int) int {
	if len(nums) == 0 {
		return 0
	}

	maxSoFar := nums[0]
	maxEndingHere := nums[0]

	for i := 1; i < len(nums); i++ {
		maxEndingHere = max(nums[i], maxEndingHere+nums[i])
		maxSoFar = max(maxSoFar, maxEndingHere)
	}

	return maxSoFar
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

func main() {
	var n int
	fmt.Print("Enter the length of the array: ")
	fmt.Scan(&n)

	nums := make([]int, n)
	fmt.Printf("Enter %d integers for the array: ", n)
	for i := 0; i < n; i++ {
		fmt.Scan(&nums[i])
	}

	result := kadanesAlgorithm(nums)
	fmt.Println("The maximum sum of a contiguous subarray is:", result)
}

File: KadaneAlgorithm/kadane.py
def maxSubArraySum(a):
    if len(a) == 0:  # Check if array is empty
        return 0
    else:
        size = len(a)
        max_so_far = a[0]
        curr_max = a[0]

        for i in range (1 , size):
            curr_max = max (a[i], curr_max+a[i])
            max_so_far = max (max_so_far,curr_max)

        return max_so_far

size = int(input("Enter the number of elements: "))
arr = []
for i in range(size):
    num = int(input("Enter element "+str(i+1)+": "))
    arr.append(num)

print ("Maximum contiguous sum is", maxSubArraySum(arr))

